### PR TITLE
Replacing term View by term ddView.

### PR DESCRIPTION
**Pull request by walter-weinmann**
DDerl views represent (offloaded) database views which consist of

- a view definition consisting of a select query with
- - -  expression list and optional aliases (like DB views)
- - -  optional join conditions (like DB views)
- - - where clause and order by (like DB views) 
- additional presentation information (window size and position, column widths)
- DDerl Views serve as building blocks for Dashboards (layout of Dderl Views on a screen).

### CODE REVIEW

**Comment by acautin** [comment]
@walter-weinmann please update the brach with latest changes to merge this.

**Comment by walter-weinmann** [comment]
Should be ok now!

On 11 April 2016 at 08:23, Agustin <notifications@github.com> wrote:

> @walter-weinmann <https://github.com/walter-weinmann> please update the
> brach with latest changes to merge this.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/K2InformaticsGmbH/dderl/pull/203#issuecomment-208181567>
>



-- 

___________________________________________________________________________
*Walter Weinmann*
___________________________________________________________________________
Obertorplatz 4
CH-4130 Rheinfelden
Tel +41 (0)61 841 06 10
___________________________________________________________________________
Schulstrasse 1
CH-6037 Root
Tel +41 (0)41 530 39 70
___________________________________________________________________________

File: src/oci_adapter.erl [hunk 1]
```diff
@@ -35,7 +35,7 @@ init() ->
         { <<"Remote Tables">>
         , <<"select concat(OWNER,concat('.', TABLE_NAME)) as QUALIFIED_TABLE_NAME from ALL_TABLES where OWNER=user order by TABLE_NAME">>
         , [] },
-        { <<"Remote Views">>
+        { <<"Remote ddViews">>
```
Comment (acautin) [inline src/oci_adapter.erl]: @walter-weinmann in this case the name Views is correct as it is referring to oracle views.

**Comment by walter-weinmann** [comment]
This change is done, but what about

**dderl.table.js:**

```
_deleteView: function() {
    if(("All ddViews" === this.options.title) || ("Remote Tables" === this.options.title)) {
        alert_jq("Error: The ddView '" + this.options.title + "' may not be deleted");
    } else if(this._viewId) {
        var self = this;
        var viewName = self.options.title;
        confirm_jq({title: "Confirm delete view " + viewName, content: ''},
            function() {
                console.log("deleting a view " + self._viewId + " with name " + viewName);
                var delView = {view_op : {operation : "delete", view_id : self._viewId, newname : ""}};
                self._ajax('view_op', delView, 'view_op', 'opViewResult');
            });
    } else {
        alert_jq("Error: \"" + this.options.title + "\" is not a view and therefore may not be deleted here!");
    }
},
```
Is here the the restriction ` } else if(this._viewId) { `still ok ?

**Comment by acautin** [comment]
@walter-weinmann that will work now as it is the same variable we are setting on

```
_openView: function(viewResult) {
        this._cmd    = viewResult.content;
        this._stmt   = viewResult.statement;
        this._conn   = viewResult.connection;
        this._viewId = viewResult.view_id;
        ...
```

Otherwise you will need to change all references to viewId.

**Comment by walter-weinmann** [comment]
I only wondered about the different handling with `_renameView` and `_deleteView`.

**Comment by acautin** [comment]
@walter-weinmann yes it seems correct, as we can't delete a view that doesn't exist, however it was asked that we set the table title even if the view doesn't exist yet. That's why there is a difference in handling.

Merging this as it looks ok.

**Pull request merged**